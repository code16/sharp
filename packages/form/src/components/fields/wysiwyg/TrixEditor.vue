<template>
    <div class="SharpTrix" :class="{ 'SharpTrix--read-only': readOnly }">
        <div class="SharpModule__inner">
            <input :id="inputId" :value="text" type="hidden">
            <trix-toolbar v-if="toolbar" class="SharpModule__header" :id="toolbarId">
                <trix-custom-toolbar :toolbar="toolbar" />
            </trix-toolbar>
            <trix-editor class="SharpModule__content"
                :input="inputId"
                :toolbar="toolbarId"
                :placeholder="placeholder"
                :style="{ height: `${height}px`, maxHeight:`${height}px` }"
                @trix-change="handleChanged"
                ref="trix"
            ></trix-editor>
        </div>
    </div>
</template>

<script>
    import TrixCustomToolbar from './TrixCustomToolbar.vue';

    import localize from '../../../mixins/localize/editor';

    export default {
        name:'SharpTrix',

        mixins: [ localize({ textProp:'text' }) ],

        components: {
            TrixCustomToolbar
        },
        props: {
            value: Object,
            toolbar: Array,
            height: {
                type: Number,
                default: 250
            },
            placeholder: String,
            readOnly: Boolean,
            uniqueIdentifier: String,

        },
        watch: {
            locale() {
                this.localized && this.$refs.trix.editor.loadHTML(this.text);
            }
        },
        computed: {
            inputId() {
                return `trix-input-${this.uniqueIdentifier}`;
            },
            toolbarId() {
                return `trix-toolbar-${this.uniqueIdentifier}`;
            },
            text() {
                return this.localized ? this.localizedText : this.value.text;
            }
        },
        methods: {
            handleChanged(event) {
                this.$emit('input', this.localizedValue(event.target.value));
            }
        },
        created() {
            window.Trix.config.toolbar.getDefaultHTML = () => '';
        }
    }
</script>