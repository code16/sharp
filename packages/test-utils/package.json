{
    "name": "@sharp/test-utils",
    "private": true,
    "main": "src/index.js",
    "version": "1.0.0"
}