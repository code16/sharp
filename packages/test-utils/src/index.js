export { default as MockInjections } from './MockInjections';
export { default as MockI18n } from './MockI18n';
export { default as QueryComponent } from './QueryComponent';
export { default as MockTransitions } from './MockTransitions';
export { mockSFC, unmockSFC } from './mockSFC';
export { mockProperty, unmockProperty, setter } from './mock-utils';
export { nextRequestFulfilled } from './moxios-utils';

export function wait(delay) {
    return new Promise(resolve => {
        setTimeout(resolve, delay);
    });
}