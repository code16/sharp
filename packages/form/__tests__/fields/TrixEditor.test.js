import { shallowMount } from '@vue/test-utils';
import TrixEditor from '../../src/components/fields/wysiwyg/TrixEditor.vue';


describe('trix-editor',()=>{
    let wrapper;

    window.Trix = require('trix');

    beforeEach(() => {
        wrapper = shallowMount(TrixEditor, {
            propsData: {
                uniqueIdentifier: 'WYSIWYG_TEST',
                value: {
                    text:'<div>Wysiwyg text</div>'
                },
                toolbar: [
                    'bold', 'italic', '|', 'link'
                ],
                placeholder: 'placeholder',
                height: 199,
            },
            attachToDocument: true
        });
    });

    test('can mount TrixEditor', ()=>{
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('can mount "read only" TrixEditor', ()=>{
        wrapper.setProps({ readOnly:true });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('emit input', ()=>{
        wrapper.find('trix-editor').trigger('trix-change');
        expect(wrapper.emitted().input).toEqual([[{
            text:'<div>Wysiwyg text</div>'
        }]])
    });
});