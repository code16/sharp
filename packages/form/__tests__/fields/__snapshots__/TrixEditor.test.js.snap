// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trix-editor can mount "read only" TrixEditor 1`] = `
<div class="SharpTrix SharpTrix--read-only">
	<div class="SharpModule__inner">
		<input id="trix-input-WYSIWYG_TEST" type="hidden" value="<div>Wysiwyg text</div>">
		<trix-toolbar id="trix-toolbar-WYSIWYG_TEST" class="SharpModule__header">
			<trix-custom-toolbar-stub toolbar="bold,italic,|,link"></trix-custom-toolbar-stub>
		</trix-toolbar>
		<trix-editor contenteditable="" role="textbox" input="trix-input-WYSIWYG_TEST" toolbar="trix-toolbar-WYSIWYG_TEST" placeholder="placeholder" class="SharpModule__content" style="height: 199px; max-height: 199px;"></trix-editor>
	</div>
</div>
`;

exports[`trix-editor can mount TrixEditor 1`] = `
<div class="SharpTrix">
	<div class="SharpModule__inner">
		<input id="trix-input-WYSIWYG_TEST" type="hidden" value="<div>Wysiwyg text</div>">
		<trix-toolbar id="trix-toolbar-WYSIWYG_TEST" class="SharpModule__header">
			<trix-custom-toolbar-stub toolbar="bold,italic,|,link"></trix-custom-toolbar-stub>
		</trix-toolbar>
		<trix-editor contenteditable="" role="textbox" input="trix-input-WYSIWYG_TEST" toolbar="trix-toolbar-WYSIWYG_TEST" placeholder="placeholder" class="SharpModule__content" style="height: 199px; max-height: 199px;"></trix-editor>
	</div>
</div>
`;
